### DISCUSSION TITLE
Pretty print array/json in diff

### DISCUSSION THREAD

**Discussion by mansoorkhan96** [jfcherng/php-diff#87] [Q&A]
I am using this in a Laravel/Filament plugin to show a diff between different versions of models. I am wondering if it is possible to show diff for arrays or json in a formatted / prettier way?

If it can take a string and also take a param to specify if its a json string, so that i can format and show in a nice readable way?

**Comment by jfcherng** [comment]
I don't think this library will do that, but still I suggest that you give an EXACT expected input/output. Your description is just too vague making me unwilling to assume this assume that and finally find out we are talking about different things and wasting time.

https://github.com/swaggest/json-diff

**Reply by mansoorkhan96** [reply]
Alright. yes the description is pretty vague, apologies.

Ok, think of the input as a json string and the out put to be a formatted version it. Just like how our editors format it. Think of it how github handles the diff for a json file.

Here is a screenshot from an online json viewer, where i throw the json string and it formats it and prints it form me.

<img width="1350" alt="image" src="https://github.com/user-attachments/assets/2e0e9a8f-0a7b-445e-9778-ae527292cd44">

**Reply by jfcherng** [reply]
Formatting isn't the issue. You can do that on your side beforehand. But pure string comparison may have issues.

```json
{
    "a": 1,
    "b": 2
}
```



```json
{
    "b": 2,
    "a": 1
}
```

Are these two JSON objects are the same? Yes, but they are not the same when treated as two strings. The user may or may not want to show them as the same and hence making comparing them using pure string comparison complicated.

**Reply by mansoorkhan96** [reply]
oh yes, maybe i should try to format and then pass. I will also try this.

**Comment by jfcherng** [comment]
What this lib does is basically the `diff` utility on your Linux/MacOS machine. If you can manage to use the `diff` utility to achieve your goal then most likely this lib can as well and most likely it already does as well.

This lib doesn't and won't handle any structured input. It's just pure string comparison like the `diff` utility.

**Reply by mansoorkhan96** [reply]
Understood, totally makes sense.

**Comment by jfcherng** [comment]
https://github.com/Wilfred/difftastic may help as well if you are willing to call an external 3rd-party executable.

**Reply by mansoorkhan96** [reply]
Thanks.
yes, this and i also stumbled upon a Javascript library. I am not sure what was it..

**Reply by mansoorkhan96** [reply]
found it.
https://diff2html.xyz/

**Reply by mansoorkhan96** [reply]
i have to check if it would work in my use case.
